server/src/main/java/org/opensearch/index/SegmentReplicationStatsTracker.java

```diff
@@ -28,11 +29,24 @@ public class SegmentReplicationStatsTracker {
     private final IndicesService indicesService;
     private final Map<ShardId, AtomicInteger> rejectionCount;
 
+    @Inject
```

Comment:
An instance of this class is created in Node and passed to other components.  We should bind to that instance only and remove this.

Reply:
Added in latest commit